lib/beacon/content/page.ex

```diff
@@ -99,8 +100,16 @@ defmodule Beacon.Content.Page do
   # need them to get going on the admin interface for now
   # TODO: only allow path if status = draft
   @doc false
-  def update_changeset(page, attrs) do
+  def update_changeset(page, attrs \\ %{}) do
     {extra_attrs, attrs} = Map.pop(attrs, "extra")
+    {ast, attrs} = Map.pop(attrs, "ast")
+
+    attrs =
+      if is_nil(ast) do
+        attrs
+      else
+        Map.put(attrs, :template, HeexTransformer.transform(ast))
+      end
```

Comment:
Instead of manipulating `attrs` in the changeset, it's better to move this block of code to https://github.com/BeaconCMS/beacon/blob/47490d983b99f92b893b0a9cfa55aac1b3aa3c14/lib/beacon/content.ex#L485 so the schema doesn't need to know about `ast`, it only cares about `template`.

Reply:
Done